src/task.c

```diff
@@ -380,7 +384,8 @@ task_thread_create(const char            *name,
    ret = task_create_thread_with_hooks(
       &thr, FALSE, func, arg, scratch_size, ts, hid);
    if (!SUCCESS(ret)) {
-      platform_error_log("Could not create a thread\n");
+      platform_error_log("Could not create a thread: %s\n",
+                         platform_status_to_string(ret));
```

Comment:
Added `platform_status_to_string()` to decode ret value. In the newly amended failing test case, you will now see this message:

```
Could not create a thread: Resource temporarily unavailable
```

Reply:
With the previous return of `STATUS_LIMIT_EXCEEDED` (which is mapped to `CONST_STATUS(ENOSPC)`, we would get a message `No space left on device` that is quite misleading

This new message is a bit more accurate as the limit on all threads in-use could go away as soon as one or more threads exit.